src/librustc_resolve/lib.rs

```diff
@@ -909,7 +935,11 @@ enum PathResult<'a> {
 }
 
 enum ModuleKind {
+    /// Inline `mod something { ... }`.
```

Comment:
`ModuleKind::Block` is an anonymous module, i.e. a block:
```rust
fn main() {
    fn f() {} // (1)
    {
        f(); // This resolves to (2)
        fn f() {} (2)
    }
}
```

Whether a module is inline or not has no effect on name resolution.